From b491a7d44e55271e155aeafb4b1aefe52b4dde37 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Mon, 23 Mar 2015 17:46:13 +0800
Subject: [PATCH] MLK-10436-04: arm: imx: report cpu revision of i.MX6QP as
 Rev 1.0

for i.MX6QP chip, the most of the code is same as i.MX6DQ, in kernel
code, still use cpu_is_6q() check the SOC Rev, report the chip version
as "i.MX6QP rev 1.0" makes more sense.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/mach-imx/mach-imx6q.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/mach-imx6q.c b/arch/arm/mach-imx/mach-imx6q.c
index 30a4aef..1bdeb4d 100644
--- a/arch/arm/mach-imx/mach-imx6q.c
+++ b/arch/arm/mach-imx/mach-imx6q.c
@@ -448,8 +448,11 @@ static void __init imx6q_init_machine(void)
 {
 	struct device *parent;
 
-	imx_print_silicon_rev(cpu_is_imx6dl() ? "i.MX6DL" : "i.MX6Q",
-			      imx_get_soc_revision());
+	if (cpu_is_imx6q() && imx_get_soc_revision() == IMX_CHIP_REVISION_2_0)
+		imx_print_silicon_rev("i.MX6QP", IMX_CHIP_REVISION_1_0);
+	else
+		imx_print_silicon_rev(cpu_is_imx6dl() ? "i.MX6DL" : "i.MX6Q",
+				 imx_get_soc_revision());
 
 	mxc_arch_reset_init_dt();
 
-- 
1.8.0

